composer.json

```diff
@@ -10,22 +10,35 @@
         {"name": "Roman Borschel", "email": "roman@code-factory.org"},
         {"name": "Benjamin Eberlei", "email": "kontakt@beberlei.de"},
         {"name": "Jonathan Wage", "email": "jonwage@gmail.com"},
-        {"name": "Johannes Schmitt", "email": "schmittjoh@gmail.com"}
+        {"name": "Johannes Schmitt", "email": "schmittjoh@gmail.com"},
+        {"name": "Fabio B. Silva", "email": "fabio.bat.silva@gmail.com"}
     ],
     "require": {
-        "php": ">=5.3.2",
-        "doctrine/lexer": "1.*"
+        "php":          ">=7.0.0",
+        "hoa/compiler": "3.*"
     },
     "require-dev": {
-        "doctrine/cache": "1.*",
-        "phpunit/phpunit": "4.*"
+        "doctrine/cache"                        : "1.*",
+        "phpunit/phpunit"                       : "5.*",
+        "instaclick/coding-standard"            : "~1.1",
+        "squizlabs/php_codesniffer"             : "~1.5",
+        "satooshi/php-coveralls"                : "~1.0",
+        "instaclick/object-calisthenics-sniffs" : "dev-master",
+        "instaclick/symfony2-coding-standard"   : "dev-remaster"
     },
     "autoload": {
-        "psr-0": { "Doctrine\\Common\\Annotations\\": "lib/" }
+        "psr-4": {
+            "Doctrine\\Annotations\\": "src/"
```

Comment:
couldn't we keep the same namespace to make the change less disruptive for userland (possibility to keep using the same API) ?

Reply:
I don't know all the details of this, but I agree with @stof to focus on internal changes.

I always found the global AnnotationRegistry to be ugly. Instead, the parser should produce an AST first, where annotation names are not treated as classes. Then a separate step should resolve those names as objects.

I see that the PR does this. The first step is to let the hoa parser construct an AST. Then the DocVisitor (strange name, btw) turns it into an annotation object tree. And instead of the awkward class registry, ist just checks `class_exists() || interface_exists()`. Which is ok I guess.

This part looks reasonable, so this should be the focus - imo.